src/DataModel/IMetadataProvider.ts

```diff
@@ -0,0 +1,30 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import "whatwg-fetch";
+import {IData} from "./IData";
+
+/**
+ * @interface Describes an abstract meta-data providing facility which translates from raw @link Response to an abstract data model.
+ */
+export interface IMetadataProvider
```

Comment:
I agree with the general idea of separation. Although I'm anxious to see how framing alone will be enough. I've been there too and I had to resort to actual triple processing as an intermediate processing step